### PR TITLE
Add `max_part_size` parameter to `MultiPartParser`

### LINKED ISSUE #2785: MultiPartException:  Part exceeded maximum size 1M
It came from [here](https://github.com/encode/starlette/commit/fd038f3070c302bff17ef7d173dbb0b007617733#diff-59eaf7e36ef8b186735f99bca714455f02f04cee26e014dc89120417996d35b2R121).  @ version [0.41.3](https://github.com/encode/starlette/releases/tag/0.41.3)

Need to make this variable customizable.

### PR DESCRIPTION
**Pull request by iudeen**
Fixes #2785 

Also addresses issue discussed in [FastAPI Repo](https://github.com/fastapi/fastapi/discussions/12961#discussioncomment-11336476).

### CODE REVIEW

File: starlette/formparsers.py [hunk 1]
```diff
@@ -122,16 +122,15 @@ async def parse(self) -> FormData:
...
-
     def __init__(
         self,
         headers: Headers,
         stream: typing.AsyncGenerator[bytes, None],
         *,
         max_files: int | float = 1000,
         max_fields: int | float = 1000,
+        max_file_size: int = 1024 * 1024,  # 1MB
+        max_part_size: int | float = 1024 * 1024,  # 1MB
```
Comment (iudeen) [inline starlette/formparsers.py]: Should we remove defaults here and keep the default at top most function?

File: starlette/formparsers.py [hunk 2]
```diff
@@ -148,6 +147,8 @@ def __init__(
         self._file_parts_to_write: list[tuple[MultipartPart, bytes]] = []
         self._file_parts_to_finish: list[MultipartPart] = []
         self._files_to_close_on_error: list[SpooledTemporaryFile[bytes]] = []
+        self.max_file_size = max_file_size
+        self.max_part_size = max_part_size
```
Comment (Kludex) [inline starlette/formparsers.py:151]: Let's just make the max_part_size configurable in this PR. Not both.

File: starlette/formparsers.py (hunk 1 above)
Comment (Kludex) [reply starlette/formparsers.py]: This is okay.

File: starlette/requests.py [hunk 3]
```diff
@@ -278,9 +285,15 @@ async def _get_form(self, *, max_files: int | float = 1000, max_fields: int | fl
         return self._form
 
     def form(
-        self, *, max_files: int | float = 1000, max_fields: int | float = 1000
+        self,
+        *,
+        max_files: int | float = 1000,
+        max_fields: int | float = 1000,
+        max_part_size: int | float = 1024 * 1024,
```
Comment (Kludex) [inline starlette/requests.py]: This doesn't need to be a float.

**Review by Kludex — CHANGES_REQUESTED** [review]

File: starlette/requests.py (hunk 3 above)
Comment (iudeen) [reply starlette/requests.py]: Ah, i just followed other params. Will change them all.

Comment (Kludex) [reply starlette/requests.py]: No, the others allow float.

File: tests/test_formparsers.py [hunk 4]
```diff
@@ -104,10 +104,12 @@ async def app_read_body(scope: Scope, receive: Receive, send: Send) -> None:
     await response(scope, receive, send)
 
 
-def make_app_max_parts(max_files: int = 1000, max_fields: int = 1000) -> ASGIApp:
+def make_app_max_parts(
+    max_files: int = 1000, max_fields: int = 1000, max_file_size: int = 1024 * 1024, max_part_size: int = 1024 * 1024
```
Comment (iudeen) [inline tests/test_formparsers.py]:
```suggestion
    max_files: int = 1000, max_fields: int = 1000, max_part_size: int = 1024 * 1024
```

File: starlette/requests.py (hunk 3 above)
Comment (iudeen) [reply starlette/requests.py]: How'd files and fields be fraction? Shouldn't it be a whole number always? I'm just curious.

Comment (Kludex) [reply starlette/requests.py]: Because they accept `float("inf")`.

**Review by Kludex — APPROVED** [review]
Perfect! Thanks. :)

**Pull request merged**